dfget/core/uploader/peer_server_executor.go

```diff
@@ -70,6 +72,12 @@ type peerServerExecutor struct {
 var _ PeerServerExecutor = &peerServerExecutor{}
 
 func (pe *peerServerExecutor) StartPeerServerProcess(cfg *config.Config) (port int, err error) {
+	fileLock := fileutils.NewFileLock(filepath.Dir(cfg.RV.MetaPath))
```

Comment:
Is a simple mutex lock enough? Do you mind share more insights on why a file lock is required?

Reply:
Because dfget server processes are created by different dfget processes.

Reply:
Oh, I see, thanks!